pkg/server/commands/list_objects.go

```diff
@@ -167,7 +182,7 @@ func (q *ListObjectsQuery) evaluate(
 	req listObjectsRequest,
 	resultsChan chan<- ListObjectsResult,
 	maxResults uint32,
-	resolutionMetadata *reverseexpand.ResolutionMetadata,
+	resolutionMetadata *ResolutionMetadata,
```

Comment:
(nit non-blocking) Opinion piece here - resolution metadata is trailer metadata included and passed along as part of the request. For uniformity, dispatch composition should be consistent between ListObjects and Check, so in the near term I'd like to see this trailer metadata included in the request and not a tag-along as a parameter in the function signature. It's inconsistent.

Reply:
> in the near term I'd like to see this trailer metadata included in the request

the response you mean? since the metadata applies to the response, not the request

but i agree..

the problem is how are we going to implement it? with the streaming variant this becomes complicated 🤔

Reply:
In the request 🤔 I don't understand why should it be part of the request.

Reply:
Yeah, you guys get my drift 😄 . Sorry, it's late afternoon on a Friday, haha.